qiskit/circuit/library/n_local/evolved_operator_ansatz.py

```diff
@@ -127,9 +133,13 @@ def evolved_operator_ansatz(
 
     # fast, Rust-path
     if (
-        flatten is not False  # captures None and True
+        num_operators > 0
+        and flatten is not False  # captures None and True
         and evolution is None
-        and all(isinstance(op, SparsePauliOp) for op in operators)
+        and all(
+            isinstance(op, SparsePauliOp) or isinstance(op, SparseObservable)
```

Comment:
You can merge two isinstance checks as 
```suggestion
            isinstance(op, (SparsePauliOp, SparseObservable))
```
(this also occurs somewhere else in the code)